src/core/common/notifier.hpp

```diff
@@ -187,7 +187,8 @@ class Notifier : public InstanceLocator
     enum
     {
         kMaxExternalHandlers   = OPENTHREAD_CONFIG_MAX_STATECHANGE_HANDLERS,
-        kFlagsStringBufferSize = 128,
+        kFlagsStringLineLimit  = 70,
+        kFlagsStringBufferSize = kFlagsStringLineLimit + 25,
```

Comment:
How do we ensure we do not exceed the magic number 25?

Reply:
Technically we cannot ensure that it won't exceed. 
If the returned string from `FlagToString()` for a flag happens to be larger than this threshold and we are at end of a log line, then the string name for that flag would be clipped (to 25 chars). 

This threshold should technically be larger than max strlen of any string returned by `FlagToString()`.  I chose 25 to be safe. 

I can define a new `enum` as `kMaxFlagNameLength` to may be make it a bit cleaner?

Reply:
Yes, and adding a comment somewhere to make clear that strings in `Notifier::FlagToString()` should not exceed `kMaxFlagNameLength`.